src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,255 @@
+
+    >>> _parse_version("23.8.0")
+    (23, 8, 0)
+    >>> _parse_version("24.0.0")
+    (24, 0, 0)
+    >>> _parse_version("invalid_version")
+    Traceback (most recent call last):
+    ...
+    click.exceptions.ClickException: Invalid version format found
+    """
+    try:
+        return tuple(int(part) for part in version.split("."))
+    except ValueError as e:
+        _logger.info("Failed to parse version.", exc_info=True)
+        raise click.ClickException(f"Invalid version format found:{str(e)}") from e
+
+
+def _get_daqmx_installed_version() -> Optional[str]:
+    """
+    Check for existing installation of NI-DAQmx.
+
+    """
+    try:
+        _logger.debug("Reading the registry entries to get installed DAQmx version")
+        with winreg.OpenKeyEx(
+            winreg.HKEY_LOCAL_MACHINE,
+            r"SOFTWARE\National Instruments\NI-DAQmx\CurrentVersion",
+            0,
+            winreg.KEY_READ | winreg.KEY_WOW64_32KEY,
+        ) as daqmx_reg_key:
+            product_name = winreg.QueryValueEx(daqmx_reg_key, "ProductName")[0]
+            product_version = winreg.QueryValueEx(daqmx_reg_key, "Version")[0]
+
+        if product_name == "NI-DAQmx":
+            _logger.info(
+                "Found registry entries for Product Name: %s and version %s",
+                product_name,
+                product_version,
+            )
+            return product_version
+    except FileNotFoundError:
+        _logger.info("No existing NI-DAQmx installation found.", exc_info=False)
+        return None
+    except PermissionError as e:
+        _logger.info("Failed to read the registry key.", exc_info=True)
+        raise click.ClickException(
+            f"Permission denied while trying to read the registry key"
+        ) from e
+    except OSError as e:
+        _logger.info("Failed to read the registry key.", exc_info=True)
+        raise click.ClickException(
+            f"An OS error occurred while trying to read the registry key"
+        ) from e
+
+
+# Creating a temp file that we then close and yield - this is used to allow subprocesses to access
+@contextlib.contextmanager
+def _multi_access_temp_file(*, suffix: str = ".exe", delete: bool = True) -> Generator[str, None, None]:
+    """
+    Context manager for creating a temporary file.
+
+    """
+    try:
+        temp_file = tempfile.NamedTemporaryFile(suffix=suffix, delete=False, mode="w")
+        temp_file.close()
+        _logger.debug("Created temp file: %s", temp_file.name)
+    except Exception as e:
+        _logger.info("Failed to create temporary file.", exc_info=True)
+        raise click.ClickException(
+            f"An error occurred while trying to to create temporary file:{str(e)}"
+        ) from e
+
+    try:
+        yield temp_file.name
+    finally:
+        if delete:
+            try:
+                os.unlink(temp_file.name)
+                _logger.debug("Deleting temp file: %s", temp_file.name)
```

Comment:
It is customary to log "Doing X" before doing X, not after. That way, if X hangs or raises an error, you know what it was trying to do.
```suggestion
                _logger.debug("Deleting temp file: %s", temp_file.name)
                os.unlink(temp_file.name)
```